test/sql/pg_dump.sql

```diff
@@ -181,24 +189,29 @@ SELECT get_sqlstate('SELECT timescaledb_post_restore()');
 
 drop function get_sqlstate(TEXT);
 
-\c postgres :ROLE_SUPERUSER
---need to shutdown workers to use db as template
---<exclude_from_test>
-SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = :'TEST_DBNAME';
---</exclude_from_test>
--- there should be no sessions for that database
--- columns are explicit so the output is the same on PG12 and PG13 since PG13 has extra column leader_pid
-SELECT datid,datname,pid,usesysid,usename,application_name,client_addr,client_hostname,client_port,backend_start,xact_start,query_start,state_change,wait_event_type,wait_event,state,backend_xid,backend_xmin,query,backend_type FROM pg_stat_activity WHERE datname = :'TEST_DBNAME';
-
-CREATE DATABASE db_dump_error WITH TEMPLATE :TEST_DBNAME;
-
---now test functions for permission errors
-\c  db_dump_error :ROLE_DEFAULT_PERM_USER_2
-\set ON_ERROR_STOP 0
+-- Check that the extension can be copied from an existing database
+-- without explicitly installing it. Stop background workers since we
+-- cannot have any backends connected to the database when cloning it.
+\c :TEST_DBNAME :ROLE_SUPERUSER
 SELECT timescaledb_pre_restore();
```

Comment:
```suggestion
```
This shouldnt be needed here.

Reply:
This is used to stop the background worker threads before waiting for them to terminate.